0000-template-rfc.md

```diff
@@ -0,0 +1,57 @@
+- Feature Name: (fill me in with a unique ident, my_awesome_feature)
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: (leave this empty)
+- IPFS Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+One paragraph explanation of the feature.
+
+# Motivation
+[motivation]: #motivation
+
+Why are we doing this? What use cases does it support? What is the expected outcome?
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Explain the proposal as if it was already included in the language and you were teaching it to another contributor on this project. That generally means:
+
+- Introducing new named concepts.
+- Explaining the feature largely in terms of examples.
+- Explaining how programmers and other contributors should *think* about the feature, and how it should impact the way they use the software or protocols being modified. It should explain the impact as concretely as possible.
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe the differences between teaching this to existing contributors and new contributors.
+
+For implementation-oriented RFCs (e.g. for compiler internals), this section should focus on how compiler contributors should think about the change, and give examples of its concrete impact. For policy RFCs, this section should provide an example-driven introduction to the policy, and explain its impact in concrete terms.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+This is the technical portion of the RFC. Explain the design in sufficient detail that:
+
+- Its interaction with other features is clear.
+- It is reasonably clear how the feature would be implemented.
+- Corner cases are dissected by example.
+
+The section should return to the examples given in the previous section, and explain more fully how the detailed proposal makes those examples work.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+# Rationale and alternatives
+[alternatives]: #alternatives
+
+- Why is this design the best in the space of possible designs?
+- What other designs have been considered and what is the rationale for not choosing them?
```

Comment:
The guidance questions in this section (particularly the first two) have an air of "make a strong case for why your design is good, and a neutral/weak case for the others". Might be useful to frame it more as "make a strong case for why your design is good, a similarly strong case for why the primary alternatives are good, *then* explain where the differences are and why yours is better". The idea would be to try to get the writer to steelman the alternatives and avoid strawmanning them ([definitions of those terms](https://wiki.lesswrong.com/wiki/Steel_man)), which could also help them think through/motivate their design even more.

For example, the second question could be split into two questions like "What are the best alternatives to this design, and why are they the best alternatives?" and "What is the rationale for choosing your design over the alternatives? What qualities do the alternatives have that your design does not?". That way, they argue for their design, the alternatives, and finally the differences (including the ones that make the alternatives better).

Reply:
This sounds like something you could put in a blog post. It's overly specific for a document that's meant to define the RFC process in a lightweight way. These should be simple guidelines, not a full-blown rhetoric lesson. Maybe you could use this rhetorical approach in an RFC that you submit and then write a blog post explaining what you did and how it made the RFC better?

Reply:
Gotcha, yeah I can see why it'd make sense to not introduce a change like this -- opens the door to a lot of potential 'rhetoric tips' that would dilute the simple guidelines. Also the question in the previous section, 'Why should we *not* do this?' is a good step in the same direction, not sure whether I considered that last time.

That's a good idea on the blog post, and would be fun to write. I'll keep that in mind.